packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -354,7 +354,7 @@ const SampleFinderSamples: FC<SampleFinderSamplesProps> = memo(props => {
 
     if (errors) return <Alert>{errors}</Alert>;
 
-    if (!queryConfigs) return <LoadingSpinner />;
+    if (!queryConfigs || sampleTypeNames.length == 0) return <LoadingSpinner />;
```

Comment:
can sampleTypeNames legitimately be empty? I know it's a crazy scenario that anyone would go to sample finder before there is even sample type....

Reply:
I guess that is technically possible. You could define source types but no sample types. If we start out with sampleTypeNames undefined that should allow us to distinguish the "haven't finished getting all types" from "haven't tried" states.